doc/tfsformat.rst

```diff
@@ -0,0 +1,142 @@
+    $                   %d                   %s                  %le
+                         1            SOMETHING              4.51395
+                     21345         "WITH SPACE"             123.4825
+
+
+Types and Identifiers
+---------------------
+
+Type identifiers are specific strings that indicate as which type to interpret the given data as, in the individual entries of the `Headers`_ as well as for the columns of the `Table`_.
+The following type identifiers and their corresponding Python types are accepted and used by `tfs-pandas`:
+
+================  ======================  =============== =========================
+Type Indentifier  Associated Python Type          Example               Accepted by
+================  ======================  =============== =========================
+%s                                string        `%s name`  ``MAD-X`` and ``MAD-NG``
+%bpm_s                            string    `%bpm_s name`  ``MAD-X`` and ``MAD-NG``
+%d                        64-bit integer         `%d 174`  ``MAD-X`` and ``MAD-NG``
+%hd                       64-bit integer        `%hd 174`  ``MAD-X`` and ``MAD-NG``
+%f                          64-bit float       `%f 0.946`  ``MAD-X`` and ``MAD-NG``
+%le                         64-bit float      `%hf 0.946`  ``MAD-X`` and ``MAD-NG``
+%b                               boolean        `%b true`           Only ``MAD-NG``
+%lz                      128-bit complex   `%lz 1.4+2.6i`           Only ``MAD-NG``
+================  ======================  =============== =========================
+
+It is also possible to include the length of the value into the type identifier, as in `%10s` for a string of length 10.
+`MAD-X` and `MAD-NG` inlcude the length in their output files, yet they are not mandatory to be able to read the files back into these programms. 
+`tfs-pandas` does not write length-values out and ignores any length-values when parsing files.
+It treats the identifier as if given without, meaning that the given data does not need to comply to the given length.
+
+.. admonition:: MAD-NG Specific Types
+
+    Both boolean and complex types are specific to the ``MAD-NG`` code, and would not be accepted by ``MAD-X``.
+    Please see the :doc:`compatibility section <compatibility>` for more information.
+
+.. _tfs-pandas caveats:
+
+TFS-Pandas Caveats
+------------------
+
+The following caveats apply to the `tfs-pandas` package:
+
+- Column names should be strings.
+- No spaces should be present in column names.
+- The table should not contain duplicate indices.
+- The table should not contain duplicate columns.
+- If spaces are present in strings, they should be enclosed in either single or double quotes.
+- The table data should not contain nested structures (lists, tuples, etc.).
+- The table data should not contain non-physical values (``NaN``, ``Inf``, etc.) as they would not be read back by either ``MAD-X`` or ``MAD-NG``.
```

Comment:
Not relevant to the point you're getting across, but are we sure this is the case for MAD-NG? 
I think `nan` and `inf` might be accepted by MAD-NG - therefore it might be possible for MAD-NG to output a table with such values - untested. I can look into this, if you think this might cause issues.

Also, I assume `None` or the equivalent Lua `nil` is in the etc?